add-kml-layer-with-network-links/src/main/java/com/esri/arcgismaps/sample/addkmllayerwithnetworklinks/screens/MainScreen.kt

```diff
@@ -0,0 +1,62 @@
+/* Copyright 2024 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.addkmllayerwithnetworklinks.screens
+
+import android.app.Application
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+
+import com.arcgismaps.toolkit.geoviewcompose.SceneView
+import com.esri.arcgismaps.sample.addkmllayerwithnetworklinks.components.SceneViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+
+/**
+ * Main screen layout for the sample app
+ */
+@Composable
+fun MainScreen(sampleName: String) {
+
+    val application = LocalContext.current.applicationContext as Application
+    // create a ViewModel to handle SceneView interactions
+    val sceneViewModel = SceneViewModel(application)
```

Comment:
```suggestion
    // create a ViewModel to handle SceneView interactions
    val sceneViewModel: SceneViewModel = viewModel()
```

This lets compose handle viewmodel creation and means you don't need to handle application context yourself